### PR TITLE
[5.0] Allow onAfterDisplay event to modify view output

### PR DESCRIPTION
**Pull request by Denitz**
### Summary of Changes

Currently, the `onAfterDisplay` event via `Joomla\CMS\Event\View\DisplayEvent` doesn't allow to modify the view output, but it's logical that we should be able to modify the output.

The patch allows 3rd-party plugins to modify the view output by changing the `source` argument.

### Testing Instructions

Try to change smth via `onAfterDisplay` event.

### Actual result BEFORE applying this Pull Request

You can't modify the view output before it's optionally stored to view cache. You can only use `onAfterDispatch` event and modify the `component` document buffer but this event forces the execution of custom code even if the cached output is displayed and results into extra costs.

### Expected result AFTER applying this Pull Request

You can easily modify the view output: add your handler into any system plugin which implements `SubscriberInterface`:

```
    public static function getSubscribedEvents(): array
    {
        return [
            ...
            'onAfterDisplay' => 'onAfterDisplay',
        ];
    }

    public function onAfterDisplay(\Joomla\CMS\Event\View\DisplayEvent $e)
    {
        $e->setArgument('source', '<h1>Hello from onAfterDisplay!</h1>' . $e->getArgument('source'));
    }
```
Now we can modify the view output and add custom scripts which will next be optionally cached in view cache.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/MVC/View/HtmlView.php [hunk 1]
```diff
@@ -215,9 +215,9 @@ public function display($tpl = null)
             )
         );
 
-        $eventResult->getArgument('used', false);
+        $event->getArgument('used', false);
 
-        echo $result;
+        echo $event->getArgument('source');
```
Comment (Fedik) [inline libraries/src/MVC/View/HtmlView.php]: Please use `$event->getArgument('source', $result);`

File: libraries/src/MVC/View/HtmlView.php [hunk 2]
```diff
@@ -215,9 +215,9 @@ public function display($tpl = null)
             )
         );
 
-        $eventResult->getArgument('used', false);
+        $event->getArgument('used', false);
```
Comment (Fedik) [inline libraries/src/MVC/View/HtmlView.php]: We can remove this line, I do not see any reason why it should stay.

Comment (Denitz) [reply libraries/src/MVC/View/HtmlView.php]: Agree!

**Review by Fedik — APPROVED** [review]

**Comment by HLeithner** [comment]
Several Maintainers are not happy with this, because we already have events to modify the document. So I'm closing this for now. thanks anyway for creating this pr.

**Pull request closed without merging**